org_status/org_hosts/github.py

```diff
@@ -16,14 +18,34 @@ def __init__(self, token, group, **kargs):
         self._token = GitHubToken(token)
         self._org = GitHubOrganization(self._token, self._group)
 
-        self._status_provider = self.StatusProvider(self._group)
+        self._status_provider = []
+        for i in enumerate(self.StatusProvider):
+            self._status_provider.append(self.StatusProvider[i[0]](self._group))
 
     def process_repository(self, repo, branch='master'):
         self.print_status(repo.web_url)
 
         # reliable enough?
         repo_name = repo.web_url.split('/')[-1]
-        repo_status = self._status_provider.get_status(repo_name, branch=branch)
+        repo_status = []
+        for i in enumerate(self._status_provider):
+            repo_status.append(self._status_provider[i[0]]
+                               .get_status(repo_name,
+                                           self.HostName,
+                                           branch=branch))
+
+        # if one result is passing, return that one
+        if Status.PASSING in repo_status:
```

Comment:
* I think every OrgHost and StatusProvider should be consider a unique combination and checks should be independent of each other.
* Should probably be handled in aggregation phase.
* If not ^ for now, this should at least be a part of OrgHost.